lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
The whole point of this is that I cannot set ``galaxy_infrastructure_url`` because I don't know the IP address of the VM ahead of time. If you set ``galaxy_infrastructure_url`` - ``galaxy_infrastructure_web_port`` is ignored. Also I am not distinguishing between 80 and 443 - I am trying to force IPython to access 80 instead of 8080.